docs/glusto_framework.md

```diff
@@ -0,0 +1,199 @@
+# Gluster-block glusto test framework 
+
+Glusto is a test automation framework for a distributed systems like glusterfs.
+Glusto-tests repo provides libraries for automating test cases for glusterfs, gluster-block and other related projects.
+
+The aim of this document is to help in getting started with automating test cases for gluster-block. Therefore, this includes very basic test cases.
+
+## Installing
+
+To automate test cases using glusto, we need to install glusto and other libraries/modules specific to glusterfs and gluster-block.
+Follow this link for the setup: https://github.com/gluster/glusto-tests
+
+## Running the tests
+
+Before writing gluster-block test cases, create 3-4 virtual machines which will serve as clients and servers.
+Edit the config file tests/gluster_tests_config.yml inside glusto-tests repo with the information of the allocated machines.
+We can start writing the test cases now.
+
+The test libraries for block are under glustolibs-gluster package which we have already installed. Import these and other required libraries for writing test cases.
+Here are a few examples of how test cases can be automated for gluster-block.
+
+### Few Examples
+
+@runs_on decorator is used to define the volume types for test cases. Multiple volume types can be given at the same time.
+
+In the below example, the setUpClass method creates a glusterfs volume of type repicated using the information from the config file.
+For this example I am assuming that glusterd and gluster-blockd are already started on the servers but they can be started/restarted in the test cases too as you can see below. 
```

Comment:
For this example, we assume that the glusterd and gluster-blockd srevices are already running on the servers. However, they can be started/re-started in the tests cases as are shown in examples below.